rust/src/rfb/rfb.rs

```diff
@@ -151,8 +157,16 @@ impl RFBState {
         self.transactions.iter_mut().find(|tx| tx.tx_id == tx_id)
     }
 
-    fn parse_request(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_request(&mut self, flow: *const Flow, stream_slice: StreamSlice) -> AppLayerResult {
         // We're not interested in empty requests.
+        let input = stream_slice.as_slice();
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
```

Comment:
I think at this point in the parser we can't assume that input.len() == PDU length. input may contain a partial pdu, or more than one pdu.

Reply:
Should I replace `-1_i64` here as it's done in the telnet parser?
In other parsers like `SIP` , `DNS` and `NFS`, I see` input.length()` being passed so I am assuming its ok for those parsers but not for `rfb`?